runtime/oti/VMHelpers.hpp

```diff
@@ -2151,8 +2150,11 @@ class VM_VMHelpers
 		 * For fully STW GCs, there is no harm to scan them, but it's a waste of time since they are scanned during root scanning already.
 		 *
 		 * We don't scan currently scanned either - one scan is enough.
+		 * we don't scan the continuation object after finished - java stack does not exist.
 		 */
-		needScan = started && (NULL != continuation) && (!isContinuationMountedOrConcurrentlyScanned(continuation));
+		if ((NULL != continuation) && (!isContinuationMountedOrConcurrentlyScanned(continuation))) {
+			needScan = !J9VMJDKINTERNALVMCONTINUATION_FINISHED(vmThread, continuationObject);
+		}
```

Comment:
what was wrong with the previous version of this check

```
needScan = started && !finished && (NULL != continuation) && (!isContinuationMountedOrConcurrentlyScanned(continuation));
```

Reply:
btw, I suspect this is not quite correct order, since setting started flag should be the last thing. 
as it's now, it opens a possibility that a third party starts consuming stack by seeing started flag being set

https://github.com/eclipse-openj9/openj9/blob/39bceb2abf25ea9ab5b2abf44e20200cb7d8f8c4/runtime/vm/ContinuationHelpers.cpp#L168